test/ae_mdw_web/controllers/stats_controller_test.exs

```diff
@@ -165,4 +166,95 @@ defmodule AeMdwWeb.StatsControllerTest do
                |> json_response(200)
     end
   end
+
+  test "when interval_by = week, it returns the count of transactions for the latest weekly periods",
+       %{
+         conn: conn,
+         store: store
+       } do
+    start_interval1 = 0
+    start_interval2 = 1_000
+    start_interval3 = 2_000
```

Comment:
may we use multiples of 52 and add a test for a leap year?

Reply:
All three day/week/month implementations start counting for days/week/months since epoch, so leap years wouldn't really be an edge-case. I've added a testcase for 1972-1973 for which 1972 is a leap year anyway

Reply:
With 52 or 53 weeks, it can give a different datetime (timestamp) or date depending on the implementation.
The idea of the test would be not to rely on the implementation and check easily without external computations (on how `1_000` should be represented as date).